xonsh/history.py

```diff
@@ -115,14 +115,14 @@ def unlocked_files(self):
         for f in fs:
             try:
                 lj = lazyjson.LazyJSON(f, reopen=False)
-                if lj['locked']:
-                    continue
+                if only_unlocked:
+                    if lj['locked']:
+                        continue
                 # info: closing timestamp, number of commands, filename
                 files.append((lj['ts'][1], len(lj.sizes['cmds']) - 1, f))
                 lj.close()
             except (IOError, OSError, ValueError):
                 continue
-        files.sort()
```

Comment:
Thanks for this. I went this this. At this time I can't see the difference in sorting cmds by their timestamp in history.py for both ptk/readline. So I think for now this should be an improvement.